src/print_event_cb.c

```diff
@@ -296,11 +302,11 @@ fwrite_out_lf_fld_sep(FILE *f)
 void
 init_printing_events(void)
 {
-	static char str[] = "_----------------_----------------_";
+	static char str[] = "_------------------_------------------_";
+	char *c;
 
-	str[0]  = Args.separator;
-	str[17] = Args.separator;
-	str[34] = Args.separator;
+	while (c = strchr(str, '_'))
+		*c = Args.separator;
```

Comment:
Oh, got it, `str[]` here is of course not a string literal, but a char array initialized using a string literal. So it should end up in the good old data segment.
I hope.
ah, I fooled myself!